core/lib/spree/app_configuration.rb

```diff
@@ -394,7 +394,7 @@ def default_pricing_options
     # @!attribute [rw] order_contents_class
     # @return [Class] a class with the same public interfaces as
     #   Spree::OrderContents.
-    class_name_attribute :order_contents_class, default: 'Spree::OrderContents'
```

Comment:
Can we change the default so easily? I feel like this could be a breaking change, can't we use [versioned preferences](https://github.com/solidusio/solidus/blob/ca82fa6be975425f21db0e9a2bc158f73ca0a980/core/lib/spree/preferences/configuration.rb#L137) instead? It should be something like:

```ruby
versioned_preference :order_contents_class, :string,'Spree::OrderContents': , boundaries: { '4.3.0.dev' => 'Spree::SimpleOrderContents' }
```

Which keeps the value unchanged for stores started before 4.3.0 is released.

Reply:
The value actually stays unchanged if the store uses `solidus_legacy_promotions`: https://github.com/solidusio/solidus/blob/main/legacy_promotions/lib/solidus_legacy_promotions/engine.rb#L55-L57

I can't make it depend on just the Solidus version because `Spree::OrderContents` is effectively part of `solidus_legacy_promotions`. See the second commit in this PR (that I just added for extra clarity).

Reply:
Oh nice, I missed that. Thanks!